redhat-access/app/models/redhat_access/concerns/host_managed_extensions.rb

```diff
@@ -12,8 +12,10 @@ module ClassMethods
         def search_by_plan_id(key, operator, value)
           insights_plan_runner = ForemanAnsible::InsightsPlanRunner.new(Organization.current, value.to_i)
           hostname_rules_relation = insights_plan_runner.hostname_rules(insights_plan_runner.playbook)
-          host_ids = Host::Managed.where(:name => hostname_rules_relation.keys.map{|h| h.downcase}).pluck(:id)
-          { :conditions => " hosts.id IN(#{host_ids.join(',')})" }
+          hostnames = hostname_rules_relation.keys.map{|h| h.downcase}
+          hostnames.map! { |hname| hname.split('.').first } if Setting[:use_shortname_for_vms]
+          host_ids = Host::Managed.where(:name => hostnames).pluck(:id)
+          { :conditions => (host_ids.empty? ? ' 0 = 1' : " hosts.id IN(#{host_ids.join(',')})") }
```

Comment:
How about passing the host names directly instead of doing an extra query to get the ids?
```suggestion
          { :conditions => (hostnames.empty? ? ' 0 = 1' : " hosts.name IN(#{hostnames.join(',')})") }
```

Reply:
I guess, at least we are clear about it being strictly exact match